panic: TestSSH/ExitOnStop

Spotted in: https://github.com/coder/coder/actions/runs/6972047158/job/18973430969?pr=10845

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x3cba729]

goroutine 51993 [running]:
github.com/coder/coder/v2/coderd.(*API).watchWorkspace.func2({0x0?, 0x0?}, {0x0?, 0x0?, 0x0?})
	/home/runner/actions-runner/_work/coder/coder/coderd/workspaces.go:1222 +0x349
github.com/coder/coder/v2/coderd/database/pubsub.genericListener.send({0xc018dcb290?, 0x0?}, {0x71ee290, 0xc000138008}, {0x8cfaa28, 0x0, 0x0})
	/home/runner/actions-runner/_work/coder/coder/coderd/database/pubsub/pubsub_memory.go:18 +0x84
github.com/coder/coder/v2/coderd/database/pubsub.(*MemoryPubsub).Publish.func1()
	/home/runner/actions-runner/_work/coder/coder/coderd/database/pubsub/pubsub_memory.go:79 +0xfa
created by github.com/coder/coder/v2/coderd/database/pubsub.(*MemoryPubsub).Publish
	/home/runner/actions-runner/_work/coder/coder/coderd/database/pubsub/pubsub_memory.go:77 +0x214
```

--- a/coderd/workspaces.go
+++ b/coderd/workspaces.go
@@ -1219,7 +1219,6 @@ func (api *API) watchWorkspace(rw http.ResponseWriter, r *http.Request) {
 				Type: codersdk.ServerSentEventTypeError,
 				Data: codersdk.Response{
 					Message: "Forbidden reading template of selected workspace.",
-					Detail:  err.Error(),
 				},
 			})
 			return

